ClinvarAnnotator Code and Bug Fixes

melissayan opened the pull request.

bbimber commented:
Thanks - a couple comments though:

1) Can you revert the change to travis.yml?
2) Can you remove the broad's license from the test class?  That's my fault if I included that it my earlier tests - which i will check.
3) In ClinvarAnnotator.transferAnnotations(), can you remove those TODO notes and maybe convert that to javadoc on the method itself?  It's probably worth something capturing the intent of that method
4) Should there be an index file for ClinvarAnnotator.vcf?
5) Can you please have ClinvarAnnotatorIntegrationTest extend BaseIntegrationTest, and remove the redundant methods, including: initGenomeLocParser, fixfilePath, 
6) Also clean up / remove the commented out lines in ClinvarAnnotatorIntegrationTest, such as the FileUtils.copy, and the index creation stuff in getBaseArgs?

I think it'll be good after that.

melissayan commented:
I made the following changes as suggested:
1. Reverted change to travis.yml.
2. Removed license from test.
3. Converted notes to javadoc.  Can you see if my examples for transferAnnotations() and annotateValue() make sense?
4. Surprisingly I didn't need the index file for ClinvarAnnotator.vcf.  But I do need the index file for clinvarV2.vcf or I get this error:
A USER ERROR has occurred: Input clinvarV2.vcf must support random access to enable queries by interval. If it's a file, please index it using the bundled tool IndexFeatureFile
5. ClinvarAnnotatorTest now extends from BaseIntegrationTest and redundant methods are removed.
6. Removed unused code.

Also, I fixed the annotateValue() error. Previously not all ClinVar annotations were included because I used the incorrect ClinVar IDs. 

Please let me know if there's anything else.

Thank you
